pyteal/compiler/compiler_test.py

```diff
@@ -2193,3 +2193,448 @@ def access_b4_store(magic_num: pt.abi.Uint64, *, output: pt.abi.Uint64):
 
     with pytest.raises(pt.TealInternalError):
         pt.compileTeal(program_access_b4_store_broken, pt.Mode.Application, version=6)
+
+
+def test_router_app():
+    router = pt.Router()
+
+    router.add_bare_call(pt.Approve(), pt.OnComplete.ClearState)
+
+    @router.add_method_handler
+    @pt.ABIReturnSubroutine
+    def add(a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64) -> pt.Expr:
+        return output.set(a.get() + b.get())
+
+    @router.add_method_handler
+    @pt.ABIReturnSubroutine
+    def sub(a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64) -> pt.Expr:
+        return output.set(a.get() - b.get())
+
+    @router.add_method_handler
+    @pt.ABIReturnSubroutine
+    def mul(a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64) -> pt.Expr:
+        return output.set(a.get() * b.get())
+
+    @router.add_method_handler
+    @pt.ABIReturnSubroutine
+    def div(a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64) -> pt.Expr:
+        return output.set(a.get() / b.get())
+
+    @router.add_method_handler
+    @pt.ABIReturnSubroutine
+    def mod(a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64) -> pt.Expr:
+        return output.set(a.get() % b.get())
+
+    @router.add_method_handler
+    @pt.ABIReturnSubroutine
+    def all_laid_to_args(
+        _a: pt.abi.Uint64,
+        _b: pt.abi.Uint64,
+        _c: pt.abi.Uint64,
+        _d: pt.abi.Uint64,
+        _e: pt.abi.Uint64,
+        _f: pt.abi.Uint64,
+        _g: pt.abi.Uint64,
+        _h: pt.abi.Uint64,
+        _i: pt.abi.Uint64,
+        _j: pt.abi.Uint64,
+        _k: pt.abi.Uint64,
+        _l: pt.abi.Uint64,
+        _m: pt.abi.Uint64,
+        _n: pt.abi.Uint64,
+        _o: pt.abi.Uint64,
+        _p: pt.abi.Uint64,
+        *,
+        output: pt.abi.Uint64,
+    ):
+        return output.set(
+            _a.get()
+            + _b.get()
+            + _c.get()
+            + _d.get()
+            + _e.get()
+            + _f.get()
+            + _g.get()
+            + _h.get()
+            + _i.get()
+            + _j.get()
+            + _k.get()
+            + _l.get()
+            + _m.get()
+            + _n.get()
+            + _o.get()
+            + _p.get()
+        )
+
+    ap, csp, _ = router.build_program()
```

Comment:
Could a `compile()` method be added that combines `build_program()` with `pt.compileTeal()` ?

Reply:
this can be done by adding a method `compile_program` by combining `build_program` and `pt.compileTeal`, and I see it making compiler test syntax easier. Is there anywhere else you want to use such method?

Reply:
In the usage examples, we should encourage use of this. The pattern 

```python
router = abi.Router()

@router.Method
def func1(...):
   ...

@router.Method
def func2(...):
   ...

teal = router.compile_program()
```
should be promoted

Reply:
good point, I added the [9b1205a]([9b1205a](https://github.com/algorand/pyteal/pull/170/commits/9b1205a329d28b3c10756a984c9175eee0a32167)) commit to change accordingly.